src/utils/route/signedOutRoute.tsx

```diff
@@ -0,0 +1,19 @@
+import { useAuthStore } from '@/stores/auth';
+import type { NextComponentType } from 'next';
+import { useRouter } from 'next/router';
+
+const signedOutRoute = (Component: NextComponentType) => {
+  const SignedOut = (props: any) => {
```

Comment:
I think we already have type for component props here:
https://github.com/near/near-discovery/blob/main/src/pages/_app.tsx#L28-L30